packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -522,17 +523,26 @@ function commitHookEffectListMount(tag: number, finishedWork: Fiber) {
         if (__DEV__) {
           const destroy = effect.destroy;
           if (destroy !== undefined && typeof destroy !== 'function') {
+            let hookName;
+            if (effect.tag === (HasEffect | Layout)) {
```

Comment:
This may not be resilient to adding more flags, what about this?

```suggestion
            if ((effect.tag & Layout) !== NoFlags) {
```